### PR TITLE
build(contracts): alt &  lattice

### PR DESCRIPTION
**Pull request by aLIEzsss4**
The alt network cannot function properly because requesting a gaslimit of 0 results in errors, but lattice works. The .env needs to be modified with VITE_CHAIN_ID=4242

@noyyyy

### CODE REVIEW

File: packages/client/src/App.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const App = () => {
   return (
     <>
       {isPlay ? <AutoChess /> : <JoinGame />}
-      <div className="absolute bottom-20 right-4 grid">
```
Comment (noyyyy) [inline packages/client/src/App.tsx:23]: why buttom rather than top

File: packages/contracts/worlds.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "blockNumber": 29441422
   },
   "1129710": {
-    "address": "0x6eea611D1e9eA24e8Ce2Ad5aD5f1404B5Fe6ab6E",
-    "blockNumber": 3896268
```
Comment (noyyyy) [inline packages/contracts/worlds.json:15]: You don't need to deploy it on altlayer manually as github action will do it

**Review by noyyyy — CHANGES_REQUESTED** [review]
It's not contract address problem. It's due to viem like

```
TransactionExecutionError: An internal error was received.

URL: https://flashlayer.alt.technology/autochessia7806fd60
Request body: {"method":"eth_estimateGas","params":[{"from":"0x5Bf8C4Fb557e1C7F24c74db67E3d940B8c16904c","data":"0x00ac498a310000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000","maxFeePerGas":"0x59682f00","maxPriorityFeePerGas":"0x59682f00","nonce":"0x0","to":"0x6eea611D1e9eA24e8Ce2Ad5aD5f1404B5Fe6ab6E"}]}
 
Estimate Gas Arguments:
  from:                  0x5Bf8C4Fb557e1C7F24c74db67E3d940B8c16904c
  to:                    0x6eea611D1e9eA24e8Ce2Ad5aD5f1404B5Fe6ab6E
  data:                  0x00ac498a310000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000
  maxFeePerGas:          1.5 gwei
  maxPriorityFeePerGas:  1.5 gwei
  nonce:                 0
 
Request Arguments:
  from:   0x5Bf8C4Fb557e1C7F24c74db67E3d940B8c16904c
  to:     0x6eea611D1e9eA24e8Ce2Ad5aD5f1404B5Fe6ab6E
  data:   0x00ac498a310000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000
  nonce:  0

Details: gas required exceeds allowance 0
Version: viem@1.5.4
```

Can you have a play on your pc on altlayer?

**Comment by noyyyy** [comment]
> The alt network cannot function properly because requesting a gaslimit of 0 results in errors, but lattice works. The .env needs to be modified with VITE_CHAIN_ID=4242
> 
> @noyyyy

It's better to switch like

```
  const chainId = Number(params.get("chainId") || latticeTestnet.id);
```

Do you understand my old version?

File: packages/client/src/App.tsx (hunk 1 above)
Comment (aLIEzsss4) [reply packages/client/src/App.tsx:23]: It will cover up the coin.

Comment (noyyyy) [reply packages/client/src/App.tsx:23]: It's fine

**Review by noyyyy — COMMENTED** [review]
LGTM

**Review by noyyyy — APPROVED** [review]

**Pull request merged**